presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -266,6 +263,25 @@ default void finishDelete(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Notifies when select query starts
+     *
+     * @return serializable {@link ConnectorTableHandle}. This handle will be brought back in {@link #finishSelect(ConnectorSession, ConnectorTableHandle)} if null returned, no finishSelect will be called.
+     */
+    default ConnectorTableHandle beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
Imagine we have to establish distinct connection with unique ID to the remote DB to send data over.
We need to establish connection only once per query (that's why it's not possible to use `ConnectorPageSourceProvider` for that purpose).

Speaking of Transaction Manager, assuming I can store some information there, where's the proper place to start connection to the remote DB? And there's always a problem when auto-commit is off - I don't want to share the same connection between queries within transaction. It's even semantically wrong.